src/utilities/mapUtilities.js

```diff
@@ -6,13 +6,13 @@ const getPosition = (osmStop, gtfsStop) => {
 };
 
 const generateConnectionData = (connection, tileId) => {
-  if (connection[0].isOsm && !connection[1].isOsm) {
+  if (connection[0].stopType === 0 && connection[1].stopType !== 0) {
```

Comment:
Please use some more descriptive values instead of 0 and 1 ;)
I'd suggest creating some enum or static class with nice properties names and assign them values 0 and 1